app/DynamicsAdapter/DynamicsAdapter.Web/Services/Dynamics/DynamicService.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using DynamicsAdapter.Web.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace DynamicsAdapter.Web.Services.Dynamics
+{
+    public class DynamicService : IDynamicService
+    {
+
+        private readonly AppSettings _settings;
+        private DateTime TokenExpiry;
+        private string AccessToken;
+       
+        public DynamicService(AppSettings settings)
+        {
+            _settings = settings;
+           
+        }
+        public async Task<JObject> Get(string entity)
+        {
+            RefreshToken();
```

Comment:
Do you think we are going to be able to cache this token ? (not for this implementation, next iteration)

Reply:
i am not sure how it really works as i haven't worked with it before.. i just implemented the sample code provided. Why are we unable to cache ?